### PR TITLE
Fix "Already exists" error when sharing single repo for standards and certifications

### PR DESCRIPTION
**Pull request by pburkholder**
I thought it might be handy to have both my standards and my certifications in the same repo:, https://github.com/pburkholder/freedonia-frist, and then refer to them in my `opencontrol.yaml` as follows, from 

```yaml
dependencies:
  standards:
    - url: https://github.com/pburkholder/freedonia-frist/
      revision: master
  certifications:
    - url: https://github.com/pburkholder/freedonia-frist/
      revision: master
```

When I tried this I got the following:

```bash
$ compliance-masonry --verbose get
2016/07/07 14:18:08 Running with verbosity
2016/07/07 14:18:08 Retrieving certifications
2016/07/07 14:18:08 Retrieving standards
2016/07/07 14:18:08 Retrieving components
2016/07/07 14:18:08 Ensuring directory /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/components exists
2016/07/07 14:18:08 Attempting to copy local resource AU_policy into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/components/AU_policy
2016/07/07 14:18:08 Copying local resource AU_policy reursively into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/components/AU_policy
2016/07/07 14:18:08 Retrieving dependent certifications
2016/07/07 14:18:08 Attempting to clone {https://github.com/pburkholder/freedonia-frist/ master } into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist
2016/07/07 14:18:08 Initializing repo https://github.com/pburkholder/freedonia-frist/ into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist
2016/07/07 14:18:09 Cloning https://github.com/pburkholder/freedonia-frist/ into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist
2016/07/07 14:18:09 Checking out revision master for repo https://github.com/pburkholder/freedonia-frist/
2016/07/07 14:18:09 Retrieving certifications
2016/07/07 14:18:09 Ensuring directory /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/certifications exists
2016/07/07 14:18:09 Attempting to copy local resource /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/certifications/FredRAMP-low.yaml into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/certifications/FredRAMP-low.yaml
2016/07/07 14:18:09 Copying local resource /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/certifications/FredRAMP-low.yaml into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/certifications/FredRAMP-low.yaml
2016/07/07 14:18:09 source /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/certifications/FredRAMP-low.yaml dest /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/certifications/FredRAMP-low.yaml
2016/07/07 14:18:09 Retrieving standards
2016/07/07 14:18:09 Ensuring directory /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/standards exists
2016/07/07 14:18:09 Attempting to copy local resource /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/standards/FRIST-800-53.yaml into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/standards/FRIST-800-53.yaml
2016/07/07 14:18:09 Copying local resource /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/standards/FRIST-800-53.yaml into /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/standards/FRIST-800-53.yaml
2016/07/07 14:18:09 source /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources628003615/certifications/freedonia-frist/standards/FRIST-800-53.yaml dest /Users/pburkholder/Projects/pburkholder/freedonia-compliance/opencontrols/standards/FRIST-800-53.yaml
2016/07/07 14:18:09 Retrieving components
2016/07/07 14:18:09 Retrieving dependent certifications
2016/07/07 14:18:09 Retrieving dependent standards
2016/07/07 14:18:09 Retrieving dependent components
2016/07/07 14:18:09 Retrieving dependent standards
2016/07/07 14:18:09 Attempting to clone {https://github.com/pburkholder/freedonia-frist/ master } into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources401044675/standards/freedonia-frist
2016/07/07 14:18:09 Initializing repo https://github.com/pburkholder/freedonia-frist/ into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources401044675/standards/freedonia-frist
2016/07/07 14:18:09 Cloning https://github.com/pburkholder/freedonia-frist/ into /var/folders/hl/zf_j1bvs7_b18rj7bbsm35p00000gs/T/opencontrol-resources401044675/standards/freedonia-frist
2016/07/07 14:18:10 Checking out revision master for repo https://github.com/pburkholder/freedonia-frist/
2016/07/07 14:18:10 Retrieving certifications
Already exists
```

The error originates here: https://github.com/opencontrol/compliance-masonry/blob/master/config/common/resources/resourceGetter.go#L24-L26,

which in turn comes from `mapset.Reserve` per this PR. The `.Reserve` method doesn't seem to be used anywhere else in this project except at L24 of resourceGetter.go, and I'm not sure what error it's supposed to protect against, but it currently blocks what seems to be a valid use case.

I'm sure there's probably a better way to fix this, but I thought I'd start here.  All test still pass.

### CODE REVIEW

**Comment by pburkholder** [comment]
If this approach has any traction, I'll restore tests for case I deleted, https://github.com/opencontrol/compliance-masonry/pull/192/files#diff-db7f1552c00df796b3fef957630d007aL20

File: tools/mapset/map.go [hunk 1]
```diff
@@ -41,10 +39,6 @@ func (m *MapSet) Reserve(key string, value string) (result Result) {
 		m.mapOfSet[key] = innerSet
 	}
 
-	if m.mapOfSet[key].Has(value) {
-		result.Error = ErrAlreadyExists
```
Comment (afeld) [inline tools/mapset/map.go]: Per the description of the method, it sounds like this should return `false` but _not_ return an error.

**Comment by pburkholder** [comment]
Per @afeld's suggestion, `Reserve` returns `result.Success = false` now.

I don't know what should be returned for `result.Value` I've chosen `value` but I'm at a loss because the value is never used.  The only place the `Reserve` method is used in the code (afaict) is at https://github.com/opencontrol/compliance-masonry/blob/master/config/common/resources/resourceGetter.go#L24 and the value returned is _never_ used.

My GoLang isn't strong enough to understand what some of this code is supposed to be doing... but it at least works better for my purposes. I'd love to see this in a 1.1.2 release so https://github.com/pburkholder/freedonia-compliance/tree/pdb/17_logging would work.

P.

**Comment by jcscottiii** [comment]
We need to consider the case (it's unlikely though) that someone specifies this yaml with different revisions

```
dependencies:
  standards:
    - url: https://github.com/pburkholder/freedonia-frist/
      revision: master
  certifications:
    - url: https://github.com/pburkholder/freedonia-frist/
      revision: different-branch-but-master-contains-broken-certifications
```

**Comment by afeld** [comment]
Yeah, let's punt on the refactor to allow multiple different revisions. At the very least, maybe we can give an error if it detects that, but let's take care of that in a follow-up.

**Pull request merged**